src/backend/web/graphql/index.js

```diff
@@ -94,7 +116,24 @@ module.exports.resolvers = {
         const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
         const postIds = await getPosts(first, last);
 
-        return Promise.all(postIds.map(id => Post.byId(id)));
+        const result = await Promise.all(postIds.map(id => Post.byId(id)));
+
+        if (filter) {
+          // check if author name is equal to what we're searching for.
+          if (filter.author) {
+            return result.filter(posts => posts.author === filter.author);
+          }
+          // check if published date is between
+          if (filter.fromDate && filter.toDate) {
+            const fromDate = new Date(filter.fromDate);
+            const toDate = new Date(filter.toDate);
+            return result.filter(
+              posts => parseInt(posts.published, 10) >= fromDate && posts.published <= toDate
```

Comment:
Filter usually defines a plural form, and the callback a singuluar:

```js
posts.filter(post => ...)
```

Also, a `post` uses `Date` for `published` and `updated` (it's only a number of MS in Redis, but we convert it when you call `Post.byId`).  Also, in my PR for #631, I'm adding a way to query Redis for posts only within a date range.  It might make this easier, so you don't have to request them all, and filter.  You'd only have to filter for other fields within that bounded set of posts that meet the date range specified.